### PR TITLE
Add a test that could have caught bug #5142

### PR DESCRIPTION
**Pull request by hariombalhara**
Add a test that could have caught bug https://github.com/calcom/cal.com/issues/5142

### CODE REVIEW

File: apps/web/playwright/lib/testUtils.ts [hunk 1]
```diff
@@ -122,6 +118,17 @@ export async function bookFirstEvent(page: Page) {
   await expect(page.locator("[data-testid=success-page]")).toBeVisible();
 }
 
+export async function bookOptinEvent(page: Page) {
```
Comment (alannnc) [inline apps/web/playwright/lib/testUtils.ts:121]:
```suggestion
export async function bookOptInEvent(page: Page) {
```

File: apps/web/playwright/booking-pages.e2e.ts [hunk 2]
```diff
@@ -119,4 +120,18 @@ test.describe("pro user", () => {
     await page.goto(`/${pro.username}`);
     await bookFirstEvent(page);
   });
+
+  test("can book an event that requires confirmation and then that booking can be accepted by organizer", async ({
+    page,
+    users,
+  }) => {
+    await bookOptinEvent(page);
```
Comment (alannnc) [inline apps/web/playwright/booking-pages.e2e.ts:128]:
```suggestion
    await bookOptInEvent(page);
```

File: apps/web/playwright/booking-pages.e2e.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { expect } from "@playwright/test";
 import { test } from "./lib/fixtures";
 import {
   bookFirstEvent,
+  bookOptinEvent,
```
Comment (alannnc) [inline apps/web/playwright/booking-pages.e2e.ts:6]:
```suggestion
  bookOptInEvent,
```

File: apps/web/playwright/booking-pages.e2e.ts [hunk 4]
```diff
@@ -119,4 +120,18 @@ test.describe("pro user", () => {
...
+    users,
+  }) => {
+    await bookOptInEvent(page);
+    const [pro] = users.get();
+    await pro.login();
+
+    await page.goto("/bookings/unconfirmed");
+    await page.click('[data-testid="confirm"]');
+    // This is the only booking in there that needed confirmation and now it should be empty screen
+    await page.waitForSelector('[data-testid="empty-screen"]');
```
Comment (alannnc) [inline apps/web/playwright/booking-pages.e2e.ts:135]: Shouldn't test include expects to validate it works?

Comment (hariombalhara) [reply apps/web/playwright/booking-pages.e2e.ts:135]: It's a kind of indirect assertion, to verify the presence of an element.

Comment (zomars) [reply apps/web/playwright/booking-pages.e2e.ts:135]: Let's add an explicit expect so we don't actually wait for the timeout and fail fast.

**Review by zomars — DISMISSED** [review]

File: apps/web/playwright/booking-pages.e2e.ts (hunk 4 above)
Comment (hariombalhara) [reply apps/web/playwright/booking-pages.e2e.ts]: Sorry, I remember you fixing a lot of similar issues in my tests earlier. Fixed it now

**Review by JeroenReumkens — APPROVED** [review]

**Pull request merged**